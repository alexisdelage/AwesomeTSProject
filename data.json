{
  "numFound": 449,
  "start": 0,
  "numFoundExact": true,
  "docs": [
    {
      "key": "/works/OL27479W",
      "title": "The Two Towers",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL27495W",
      "title": "The Silmarillion",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL14933414W",
      "title": "The Fellowship of the Ring",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL262758W",
      "title": "The Hobbit",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL27516W",
      "title": "The Return of the King",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL27448W",
      "title": "The Lord of the Rings",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL27517W",
      "title": "Farmer Giles of Ham",
      "author_name": ["J.R.R. Tolkien"]
    },
    {
      "key": "/works/OL27466W",
      "title": "Unfinished Tales of N\u00famenor and Middle-earth",
      "author_name": ["J.R.R. Tolkien", "Christopher Tolkien"]
    },
    {
      "key": "/works/OL27435W",
      "title": "The Book of Lost Tales, Part One",
      "author_name": ["J.R.R. Tolkien", "Christopher Tolkien"]
    },
    {
      "key": "/works/OL27471W",
      "title": "The Children of H\u00farin",
      "author_name": ["J.R.R. Tolkien"]
    }
  ],
  "num_found": 449,
  "q": "",
  "offset": null
}
