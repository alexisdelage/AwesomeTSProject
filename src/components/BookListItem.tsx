import React from 'react';
import {Text, View, StyleSheet} from 'react-native';

interface Book {
  title: string;
  key: string;
  author_name?: string[];
}

interface BookListItemProps {
  book: Book;
}

/** A component to display a book in a list */
function BookListItem(props: BookListItemProps) {
  const {book} = props;

  return (
    <View style={styles.itemContainer}>
      <Text style={styles.itemTitle}>{book.title}</Text>
      <Text style={styles.itemDescription}>
        {book.author_name?.join(', ') || 'No authors'}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  itemContainer: {
    display: 'flex',
    justifyContent: 'center',
    margin: 10,
    marginLeft: 20,
  },
  itemTitle: {
    fontWeight: 'bold',
    color: 'black',
  },
  itemDescription: {
    color: 'grey',
  },
});

export default BookListItem;
