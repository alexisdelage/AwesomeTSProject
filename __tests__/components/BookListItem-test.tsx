import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BookListItem from '../../src/components/BookListItem';
import data from '../../data.json';

it('renders correctly', () => {
  const tree = renderer.create(<BookListItem book={data.docs[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
