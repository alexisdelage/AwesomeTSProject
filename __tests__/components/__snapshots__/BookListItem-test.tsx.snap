// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "justifyContent": "center",
      "margin": 20,
      "marginTop": 0,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontWeight": "bold",
      }
    }
  >
    The Two Towers
  </Text>
  <Text
    style={
      {
        "color": "grey",
      }
    }
  >
    J.R.R. Tolkien
  </Text>
</View>
`;
